### PR TITLE
PXC-3481: Add ability to build PXC56/57/80 from PR sources

### PR DESCRIPTION
**Pull request by Sudokamikaze**
Build URL: https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-pipeline-PXC-3481/7/
I used Kamil's PR(1409) to test it

### CODE REVIEW

**Comment by kamil-holubicki** [comment]
Hi @Sudokamikaze , Somthing's wrong. This PR is mergable to 8.0 trunk and builds and tests without problems, however, the provided link points to the job with galera tests failures having logs like
```
2021-04-07T17:34:51.525604Z 0 [ERROR] [MY-000000] [WSREP-SST] ******************* FATAL ERROR ********************** 
2021-04-07T17:34:51.526624Z 0 [ERROR] [MY-000000] [WSREP-SST] The xtrabackup version is 8.0.13.
2021-04-07T17:34:51.526642Z 0 [ERROR] [MY-000000] [WSREP-SST] xtrabackup-8.0.22 or higher is needed to perform an SST
2021-04-07T17:34:51.526657Z 0 [ERROR] [MY-000000] [WSREP-SST] /tmp/results/PXC/bin/pxc_extra/pxb-8.0/bin/xtrabackup
2021-04-07T17:34:51.526671Z 0 [ERROR] [MY-000000] [WSREP-SST] ****************************************************** 
```

Is it possible to test it by triggering mult platform job, like we do it usually?

**Comment by Sudokamikaze** [comment]
> Hi @Sudokamikaze , Somthing's wrong. This PR is mergable to 8.0 trunk and builds and tests without problems, however, the provided link points to the job with galera tests failures having logs like
> 
> ```
> 2021-04-07T17:34:51.525604Z 0 [ERROR] [MY-000000] [WSREP-SST] ******************* FATAL ERROR ********************** 
> 2021-04-07T17:34:51.526624Z 0 [ERROR] [MY-000000] [WSREP-SST] The xtrabackup version is 8.0.13.
> 2021-04-07T17:34:51.526642Z 0 [ERROR] [MY-000000] [WSREP-SST] xtrabackup-8.0.22 or higher is needed to perform an SST
> 2021-04-07T17:34:51.526657Z 0 [ERROR] [MY-000000] [WSREP-SST] /tmp/results/PXC/bin/pxc_extra/pxb-8.0/bin/xtrabackup
> 2021-04-07T17:34:51.526671Z 0 [ERROR] [MY-000000] [WSREP-SST] ****************************************************** 
> ```
> 
> Is it possible to test it by triggering mult platform job, like we do it usually?

Sure, I'll post a full param run in a while

**Comment by Sudokamikaze** [comment]
Hey @kamil-holubicki 
Here's link to param: https://pxc.cd.percona.com/job/pxc-8.0-param-PXC-3481/1/

**Comment by Sudokamikaze** [comment]
As discussed, implemented based on merge ref instead of head ref
@kamil-holubicki here's tests https://pxc.cd.percona.com/job/pxc-8.0-pipeline-PXC-3481/22/

**Review by kamil-holubicki — APPROVED** [review]
lgtm

File: pxc/jenkins/pxc56-pipeline.groovy [hunk 1]
```diff
@@ -86,6 +90,18 @@ pipeline {
                 sh '''
                     MY_BRANCH_BASE_MAJOR=5
                     MY_BRANCH_BASE_MINOR=6
+
+                    if [[ ${USE_PR} == "true" ]]; then
+                        if [ -f /usr/bin/yum ]; then
+                            sudo yum -y install jq
+                        else
+                            sudo apt-get install -y jq
```
Comment (EvgeniyPatlan) [inline pxc/jenkins/pxc56-pipeline.groovy:98]: are you sure that we don't need to update apt db here?

Comment (Sudokamikaze) [reply pxc/jenkins/pxc56-pipeline.groovy:98]: I am, it'll always use docker executor, which has yum make cache:https://github.com/Percona-Lab/jenkins-pipelines/blob/master/IaC/pxc.cd/init.groovy.d/cloud.groovy#L113

But, APT/DEB as well update their cache: https://github.com/Percona-Lab/jenkins-pipelines/blob/master/IaC/pxc.cd/init.groovy.d/cloud.groovy#L298

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**